src/test/java/uk/gov/hmcts/reform/translate/service/DictionaryServiceTest.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.translate.service;
+
+import groovy.lang.IntRange;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.translate.data.DictionaryEntity;
+import uk.gov.hmcts.reform.translate.repository.DictionaryRepository;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings({"PMD.UseConcurrentHashMap", "PMD.JUnitAssertionsShouldIncludeMessage"})
+class DictionaryServiceTest {
+
+    @Mock
+    DictionaryRepository dictionaryRepository;
+
+    @Mock
+    Iterable<DictionaryEntity> repositoryResults;
+
+    @InjectMocks
+    DictionaryService dictionaryService;
+
+    @Test
+    void shouldReturnDictionaryContents() {
```

Comment:
OK, will add - I think the integration test deals with translationPhrase is null - but you are correct, it is also required here

Will also add duplicate records tests here and in IT

Reply:
Unit tests added in  https://github.com/hmcts/ts-translation-service/pull/6/commits/46585f3e1763b52bd2709a72f4a1cfc2f0ebc77b

Unsure about the integration tests, if the `IllegalStateException` is thrown, what status code gets returned in the response?  

Would the  `PUT` endpoint perform this duplicate check and error, or would it overwrite the existing entry rather than add a duplicate ?

Reply:
I'd expect the `IllegalStateException` is likely returning a `500`, as we're not converting it anywhere. Which is OK - if we somehow found ourselves in that situation then there's nothing the consumer can do to rectify, it'd require manual intervention.
The `PUT` logic is described such that it would overwrite.